### ISSUE TITLE
Godot 3.x (Linux): Major stutter/freeze after every several minutes of gameplay

### ISSUE THREAD

**Issue by Marigem** [godotengine/godot#79816]
### Godot version

3.x

### System information

Arch linux, GTX 650, 4 × Intel® Core™ i3-3220 CPU @ 3.30GHz, 8GB of ram

### Issue description

A major stutter-freeze (lasting ~1 sec) happens after playing for a certain amount of time, varying from every several, to every dozen minutes. This happens on every project, exported or in editor, even Steam games that are made with Godot, like Primal Light.

This is what the profiler reads when it happens:

![freeze](https://github.com/godotengine/godot/assets/43501741/436eadf2-fb88-451c-9c02-b36a4125edc2)

It's not the PC or OS, because this doesn't happen in games not made with Godot. They all run smoothly, depending on how graphic intensive they are of course, but 2D games don't give me any such trouble.

### Steps to reproduce

N/A

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
> It's not the PC or OS, because this doesn't happen in games not made with Godot. 

This can't be assumed from this, can still be driver and hardware related just not affecting those cases

**Comment by Marigem** [comment]
I can safely assume this, because I've been using this PC for years and this has yet to happen with anything other than Godot. 
My windows PC also has Godot-specific performance issues, like https://github.com/godotengine/godot/issues/65649 and https://github.com/godotengine/godot/issues/71391

**Comment by AThousandShips** [comment]
What I'm saying is that if it was Godot in general, rather than specifically drivers/OS/hardware or those interacting with Godot, *everyone* would be experiencing this, and they aren't

**Comment by Zireael07** [comment]
Is VSync on or off?

**Comment by Marigem** [comment]
Vsync is on

**Comment by Calinou** [comment]
Do you play in fullscreen or in a window? Windowed mode on Windows will always suffer from more or less stuttering depending on PC specs.

**Comment by Marigem** [comment]
Yes it's in fullscreen. The profiler screenshot shows smooth gameplay until that giant spike.
